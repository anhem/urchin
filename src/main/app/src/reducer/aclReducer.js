import {Actions} from "../constants";

const aclReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.ACL.GET_ACL:
            return {...state, isFetchingAcl: true};

        case Actions.ACL.GET_ACL_SUCCESS:
            let aclGroups = toAclGroups(action.data.groups);
            let aclUsers = toAclUsers(action.data.users);
            return {...state, isFetchingAcl: false, acl: {groups: aclGroups, users: aclUsers}};

        default:
            return state;
    }
};

export default aclReducer


function toAclGroups(groups) {
    return groups.map(item => {
            return {
                groupId: item.group.groupId,
                groupName: item.group.groupName,
                read: item.aclPermissions.hasRead,
                write: item.aclPermissions.hasWrite,
                execute: item.aclPermissions.hasExecute,
            }
        }
    );
}

function toAclUsers(users) {
    return users.map(item => {
            return {
                userId: item.user.userId,
                username: item.user.username,
                read: item.aclPermissions.hasRead,
                write: item.aclPermissions.hasWrite,
                execute: item.aclPermissions.hasExecute,
            }
        }
    );
}