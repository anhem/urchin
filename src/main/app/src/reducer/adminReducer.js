import {Actions} from '../constants'

const adminReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Admin.SET_ADMIN:
            return {...state, admin: {...state.admin, [action.data.field]: action.data.value}};
        case Actions.Admin.SETUP_ADMIN:
            return {...state, isAddingAdmin: true};
        case Actions.Admin.SETUP_ADMIN_SUCCESS:
            return {...state, isAddingAdmin: false};
        default:
            return state;
    }
};

export default adminReducer