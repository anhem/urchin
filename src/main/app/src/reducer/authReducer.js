import {Actions} from '../constants'

const authReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Auth.LOGIN_INVALID:
        case Actions.Auth.LOGIN_FAILED:
        case Actions.Auth.LOGIN_SUCCESS:
        case Actions.Auth.PRECONDITION_FAILED:
        case Actions.Auth.LOGOUT_SUCCESS:
            return {...state, isLoggingIn: false, authed: action.type};
        case Actions.Auth.LOGIN:
            return {...state, isLoggingIn: true};
        default:
            return state;
    }
};

export default authReducer