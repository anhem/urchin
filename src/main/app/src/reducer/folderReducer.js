import {Actions} from "../constants";

const folderReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Folders.GET_FOLDERS:
            return {...state, isFetchingFolders: true};

        case Actions.Folders.GET_FOLDERS_SUCCESS:
            return {...state, isFetchingFolders: false, folders: action.data};

        case Actions.Folder.GET_FOLDER:
            return {...state, isFetchingFolder: true};

        case Actions.Folder.GET_FOLDER_SUCCESS:
            return {...state, isFetchingFolder: false, folder: action.data};

        case Actions.Folder.SET_FOLDER:
            return {...state, folder: {...state.folder, [action.data.field]: action.data.value}};

        case Actions.Folder.SAVE_FOLDER:
            return {...state, isSavingFolder: true};

        case Actions.Folder.SAVE_FOLDER_SUCCESS:
            return {...state, isSavingFolder: false};

        case Actions.Folder.SAVE_FOLDER_VALIDATION_ERROR:
            return {...state, fieldErrors: action.data.fieldErrors};

        default:
            return state;
    }
};

export default folderReducer