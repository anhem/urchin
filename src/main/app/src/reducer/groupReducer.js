import {Actions} from "../constants";

const groupReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Groups.GET_GROUPS:
            return {...state, isFetchingGroups: true};

        case Actions.Groups.GET_GROUPS_SUCCESS:
            return {...state, isFetchingGroups: false, groups: action.data};

        case Actions.Group.GET_GROUP:
            return {...state, isFetchingGroup: true};

        case Actions.Group.GET_GROUP_SUCCESS:
            return {...state, isFetchingGroup: false, group: action.data};

        case Actions.Group.SET_GROUP:
            return {...state, group: {...state.group, [action.data.field]: action.data.value}};

        case Actions.Group.SAVE_GROUP:
            return {...state, isSavingGroup: true};

        case Actions.Group.SAVE_GROUP_SUCCESS:
            return {...state, isSavingGroup: false};

        default:
            return state;
    }
};

export default groupReducer