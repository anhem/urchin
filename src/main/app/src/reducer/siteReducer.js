import {Actions} from "../constants";

const siteReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Site.TOGGLE_NAVIGATION:
            return {...state, navigationVisible: action.data};
        default:
            return state;
    }
};

export default siteReducer