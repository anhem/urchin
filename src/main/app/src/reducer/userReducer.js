import {Actions} from '../constants'

const userReducer = (state = {}, action) => {
    switch (action.type) {
        case Actions.Users.GET_USERS:
            return {...state, isFetchingUsers: true};

        case Actions.Users.GET_USERS_SUCCESS:
            return {...state, isFetchingUsers: false, users: action.data};

        case Actions.User.GET_USER:
            return {...state, isFetchingUser: true};

        case Actions.User.GET_USER_SUCCESS:
            return {...state, isFetchingUser: false, user: action.data};

        case Actions.User.SET_USER:
            return {...state, user: {...state.user, [action.data.field]: action.data.value}};

        case Actions.User.SAVE_USER:
            return {...state, isSavingUser: true};

        case Actions.User.SAVE_USER_SUCCESS:
            return {...state, isSavingUser: false};

        case Actions.User.SAVE_USER_VALIDATION_ERROR:
            return {...state, fieldErrors: action.data.fieldErrors};

        case Actions.User.GET_GROUPS_FOR_USER:
            return {...state, isFetchingGroupsForUser: true};

        case Actions.User.GET_GROUPS_FOR_USER_SUCCESS:
            return {...state, isFetchingGroupsForUser: false, groupsForUser: action.data};


        default:
            return state;
    }
};

export default userReducer