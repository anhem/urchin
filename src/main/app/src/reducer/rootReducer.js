import {combineReducers} from 'redux';
import userReducer from './userReducer';
import groupReducer from './groupReducer';
import folderReducer from './folderReducer';
import aclReducer from './aclReducer';
import siteReducer from './siteReducer';
import authReducer from './authReducer';
import adminReducer from './adminReducer'

const appReducer = combineReducers({
    userData: userReducer,
    groupData: groupReducer,
    folderData: folderReducer,
    aclData: aclReducer,
    siteData: siteReducer,
    authData: authReducer,
    adminData: adminReducer
});

const rootReducer = (state, action) => {
    return appReducer(state, action)
};

export default rootReducer;