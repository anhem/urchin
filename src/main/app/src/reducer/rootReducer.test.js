import reducer from './rootReducer';

describe('rootReducer', () => {

    it('initial state', () => {

        expect(reducer(undefined, {})).toMatchSnapshot();
    });

});
