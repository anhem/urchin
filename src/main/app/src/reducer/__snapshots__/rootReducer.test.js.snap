// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer initial state 1`] = `
Object {
  "aclData": Object {},
  "adminData": Object {},
  "authData": Object {},
  "folderData": Object {},
  "groupData": Object {},
  "siteData": Object {},
  "userData": Object {},
}
`;
